agent/consul/state/catalog.go

```diff
@@ -436,6 +527,12 @@ func (s *Store) ensureNodeTxn(tx *memdb.Txn, idx uint64, node *structs.Node) err
 			return nil
 		}
 		node.ModifyIndex = idx
+		if n.Node != node.Node {
```

Comment:
We are comparing pointers here, shouldn't this be a `.Equal()` ?

Reply:
My bad, these are strings

Reply:
Nope, this is a string